src/Illuminate/View/Factory.php

```diff
@@ -247,7 +261,14 @@ public function renderEach($view, $data, $iterator, $empty = 'raw|')
      */
     protected function normalizeName($name)
     {
-        return ViewName::normalize($name);
+        if (isset($this->normalizedNames[$name])) {
+            return $this->normalizedNames[$name];
+        }
+
+
+        $this->normalizedNames[$name] = ViewName::normalize($name);
+
+        return $this->normalizedNames[$name];
```

Comment:
This syntax should also be possible:

```suggestion
        return $this->normalizedNames[$name] ??= ViewName::normalize($name);
```